src/Smp.Web/ClientApp/src/app/models/message.ts

```diff
@@ -0,0 +1,34 @@
+import { User } from './user';
+
+export class Message {
+  constructor(message: Message) {
+    this.id = message.id;
+    this.senderId = message.senderId;
+    this.receiverId = message.receiverId;
+    this.createdAt = message.createdAt;
+    this.content = message.content;
+    this.conversationId = message.conversationId;
+  }
+
+  public id: number;
+  public senderId: string;
+  public receiverId: string;
+  public createdAt: Date;
+  public content: string;
+  public conversationId: string;
+
+  public getFriendlyDate(): string {
+    return new Date(this.createdAt).toLocaleString();
+  }
+}
+
+export class FriendlyMessage extends Message {
+  constructor(message: Message) {
```

Comment:
Not a high priority, but I think we might be able to reduce the need for constructors like this using rxjs's map function. Not entirely sure, though.